src/adapter/src/catalog/storage.rs

```diff
@@ -384,6 +377,9 @@ async fn migrate<S: Append>(
         (migration)(&mut txn, bootstrap_args)?;
         txn.update_user_version(u64::cast_from(i))?;
     }
+    add_new_builtin_roles_migration(&mut txn)?;
```

Comment:
I realized that it was incorrect to include these in the vec above. Doing so would only execute them once for a single upgrade. We want them run every single time to add any new objects.

Reply:
Hmm, but now every one's user versions are a bit screwed up. So I think I need to keep three empty migrations above. @mjibson, does that sound right?

Reply:
Yes.